### PR TITLE
fix: Pulsar OAuth2 Authentication Factory

### PR DESCRIPTION
**Pull request by JeffBolle**
Fixes the creation of the configured pulsar authentication factory when oauth configuration is provided.

### CODE REVIEW

File: pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarConfigTest.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
+import static org.junit.Assert.assertTrue;
+
```
Comment (KKcorps) [inline pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarConfigTest.java:35]: Remove empty line

File: pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarConfigTest.java [hunk 2]
```diff
@@ -147,7 +152,6 @@ public void testParsingConfigForOAuth() throws Exception {
   @Test
   public void testParsingConfigFailFileValidationForOAuth() throws Exception {
     String testFilePath = "file://path/to/file.json";
-
```
Comment (KKcorps) [inline pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarConfigTest.java:150]: Add this empty line back

File: pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarPartitionLevelConnectionHandler.java [hunk 3]
```diff
@@ -73,10 +73,10 @@ public PulsarPartitionLevelConnectionHandler(String clientId, StreamConfig strea
     }
   }
 
-  private Optional<Authentication> getAuthenticationFactory(PulsarConfig pulsarConfig) {
+  protected Optional<Authentication> getAuthenticationFactory(PulsarConfig pulsarConfig) {
```
Comment (KKcorps) [inline pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarPartitionLevelConnectionHandler.java:76]: If this scope is changed just for testing, can we add @VisibleForTesting annotation

**Review by KKcorps — APPROVED** [review]
Some minor comments, LGTM overall!

File: pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarPartitionLevelConnectionHandler.java (hunk 3 above)
Comment (JeffBolle) [reply pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarPartitionLevelConnectionHandler.java:76]: Yes, I just changed it to test it. Will add the annotation.

**Comment by JeffBolle** [comment]
@KKcorps thank you for the speedy review.

**Pull request merged**